Adds actionable message component and stories

danjm opened the pull request:
This PR adds a component that can be used to present a message to a user. The message can optionally give the user one or more actions to take.

Associated storybook stories are added.

Demo gif:

![actionable-message](https://user-images.githubusercontent.com/7499938/90512394-ad4acc80-e138-11ea-9dff-643df9b914ef.gif)

danjm committed:
Adds actionable message component and stories

brad-decker commented on ui/app/pages/token/actionable-message/actionable-message.js:
```diff
@@ -0,0 +1,46 @@
+ActionableMessage.propTypes = {
+  shown: PropTypes.bool,
+  message: PropTypes.string.isRequired,
+  actions: PropTypes.shape({
```
I believe this should be `PropTypes.arrayOf(PropTypes.shape(...))`

brad-decker replied:
Another Q: Are we giving too much flexibility in this API? Will we ever want *three* actions? I haven't seen the usage of it yet, or the designs, so I can't be sure... but if we will ever only show at most two actions then I think the API should be changed to 'primaryAction' and 'secondaryAction' or something to that effect and remove the array and iteration.

brad-decker replied:
Forgot to remove this proptype for actions 😬

brad-decker commented on ui/app/pages/token/actionable-message/actionable-message.stories.js:
```diff
@@ -0,0 +1,69 @@
+        {
+          label: text('First ActionLabel', 'Dismiss'),
+          onClick: action('TwoActionsWithClassNames Click 1'),
+          actionClassName: 'text-transform-uppercase',
```
I am deprecating these utility classes in favor of keeping styles inline so that editing/extending/debugging styles is easier. If you could style these a different way I would appreciate it. 

Orthogonally: Are we allowing *too much* flexibility in this API? Would we ever want these actions to be entirely controlled by the developer implementing them? If no, then I'd suggest we apply the styles a different way either using modifiers such as 'uppercase: true' that adds the appropriate class to the action.

brad-decker reviewed (changes requested):
some q's and change reqs, but the component itself looks good and works well.

danjm committed:
Code cleanup and simplification for actionable-message component

brad-decker reviewed (changes requested):
Forgot one thing, but otherwise looks good to me.

jakehaugen commented:
Couple notes:

- Can we make the padding on the top and sides 16px (equal for both)?
- Looks like we're able to choose to have the text centered or left aligned?
   - If left aligned, lets left align the actions as well. If centered, the actions can stay centered.

danjm committed:
Fix padding, alignment of actionable-message; add left aligned story

danjm commented:
@jakehaugen Updated the padding and alignment and added a left aligned example:

![updatedactionmessage](https://user-images.githubusercontent.com/7499938/90559021-924b7d00-e177-11ea-85af-7c7d78e65a19.gif)

@brad-decker There was a small js change in 90a70250f to ensure the actions div doesn't render if there are no actions, and I have removed the unused proptype now

brad-decker commented:
I'm good once @jakehaugen signs off. 👍

brad-decker commented:
@danjm just needs a rebase to fix the lint issue.
